common/sgx/eeid.c

```diff
@@ -28,26 +28,24 @@
 #include "../../host/crypto/openssl/rsa.h"
 #endif
 
-int is_eeid_base_image(const oe_sgx_enclave_properties_t* properties)
+int is_eeid_base_image(const oe_enclave_size_settings_t* sizes)
 {
-    return properties->header.size_settings.num_heap_pages == 0 &&
-           properties->header.size_settings.num_stack_pages == 0 &&
-           properties->header.size_settings.num_tcs == 1;
+    return sizes->num_heap_pages == 0 && sizes->num_stack_pages == 0 &&
+           sizes->num_tcs == 1;
```

Comment:
In the updated EEID design, the "dynamic size" EEID enclave will have NumTCS=0.

Reply:
Can you confirm that there is no code path that requires `NumTCS=1`, i.e. not for dynamic, static, or non-eeid?

Reply:
I didn't see any check requiring NumTCS>0 in current OE code.

Reply:
There isn't; the last time I tried it ran just fine with NumTCS==0. My question was about your security analysis: is there any circumstance in which NumTCS==1 is required? (I don't think so, see also other thread about entry point, but the verifier should probably reject quotes from enclaves that claim they're running with NumTCS==0.)

Reply:
As long as the EEID init code inside the enclave checks that the entry point in EEID page matches the value in enlcave property (initalized data of the base image), "dyanmic size" EEID enclave is safe to use NumTCS=0. As a defense-in-depth implementation, the verifier can reject attestation with invalid size_info in the EEID page. But the measurement check should catch any manipulation of NumTCS on "dynamic size" EEID enclave.

Reply:
Great, thanks for clarifying. I still think we don't need the entry point in the enclave properties, but I'll leave that discussion in the other thread.

Reply:
For a base image with NumTCS=0, we do need the entry point as part of the base image measurement and require the EEID init code to check eeid.entry_point matches the entry point info in the base image.